File: EASY/Smallest window containing 0, 1 and 2/solution.cpp
class Solution {
public:
    int smallestSubstring(string s) {
        int n = s.size();
        
        int c0 = 0, c1 = 0, c2 = 0;
        int l = 0, ans = INT_MAX;
        
        for(int r = 0; r < n; r++) {
            if(s[r] == '0') c0++;
            else if(s[r] == '1') c1++;
            else c2++;
            
            while(c0 > 0 && c1 > 0 && c2 > 0) {
                ans = min(ans, r - l + 1);
                
                if(s[l] == '0') c0--;
                else if(s[l] == '1') c1--;
                else c2--;
                
                l++;
            }
        }
        
        return ans == INT_MAX ? -1 : ans;
    }
};

File: EASY/Smallest window containing 0, 1 and 2/solution.java
class Solution {
    public int smallestSubstring(String S) {
        
        int zero=-1,one=-1,two=-1,res,e,s;
        int len=(int)1e9;
        
        for(int i=0;i<S.length();i++)
        {
            if(S.charAt(i)=='0')
               zero=i;
            else if(S.charAt(i)=='1')
               one=i;
            else
               two=i;
               
            if(zero!=-1 && one!=-1 && two!=-1)
            {
                e=Math.max(zero,Math.max(one,two));
                s=Math.min(zero,Math.min(one,two));
                len=Math.min(len,e-s+1);
            }
        }
        res=(len>=1e9)?-1:len;
        return res;
        
    }
};